src/main/java/com/non/model/User.java

```diff
@@ -0,0 +1,62 @@
+package com.non.model;
+
+import jakarta.persistence.*;
+import lombok.Data;
+
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.UUID;
+
+
+@Entity
+@Data
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+    private Timestamp create_time;
+    private Timestamp update_time;
+    private Integer created_by;
+    private Integer updated_by;
+
+    @Column(unique = true, nullable = false)
+    private String username;
+    private String first_name;
+    private String last_name;
+
+    @Column(unique = true, nullable = false)
+    private String email;
+
+    @Column(columnDefinition = "boolean default false")
+    private boolean is_deleted;
+
+    @Column(columnDefinition = "boolean default true")
+    private boolean active;
+
+    @Enumerated(EnumType.STRING)
+    private ROLE role;
+
+    public enum ROLE {
+        ADMIN,
+        MENTOR,
+        MENTEE
+    }
+
+    public User() {
+    }
+
+    public User(String username, String first_name, String last_name, String email, ROLE role) {
```

Comment:
just in case we are sending role, for in case we want to create a mentor directly simply pass the role too, else just pass the relevant details, default role as mentee will be assigned.

Reply:
@sanskar-19 use builder pattern.